### DISCUSSION TITLE
Initialising Multiple RAMs

### DISCUSSION THREAD

**Discussion by martandrMC** [hneemann/Digital#1077] [Q&A] [answered]
So I've been trying to marry two projects together: A processor, and a video interface. Both have RAM blocks inside sub-circuits. Those sub-circuits get instantiated in one top level project file. What I would like to do, but seemingly can't find a way to, is have the RAM of the video part be initialized (with something like a default character set) separately from the program memory of the processor.

What I would have most hoped had worked was setting both sub-circuits to "Preload program memory at startup" and have the RAMs inside those sub-circuits marked as program memory and getting initialized from two separate files.

Is there any way to use multiple files for RAM initialization in general or can I only use the one file defined in the top level and needing to multiplex the data into it?

**Comment by hneemann** [comment, marked as answer]
It is not possible to initialize more than one RAM when starting the simulation. 
Compared to a real setup this is quite good, because in reality you can't initialize rams at all.

In a graphics card you would store the character set in a ROM anyway. 
Otherwise you could not display anything on the screen at boot time.

**Reply by martandrMC** [reply]
> Compared to a real setup this is quite good, because in reality you can't initialize rams at all.

I plan transition from Digital to an FPGA, in which I have the ability to initialize any block ram from a file (with which I will implement the VRAM and a small bootstrap code memory). If the only option was using external memory chips like the FPGA's SDRAM or SRAM chips in a full size build then yeah I agree with that, but not in my case.

> In a graphics card you would store the character set in a ROM anyway.

I want to put my character set in RAM because I am building a tilemap video adapter which also has bitmap modes. I've even had to even split my VRAM in two to get the correct access patterns from it.

In any case, you mentioned that it is not possible after all. Thanks for clarifying.
